### ISSUE TITLE
Make it work on a BeagleBone

### ISSUE THREAD

**Issue by workhorsy** [workhorsy/py-cpuinfo#15]
We can get the Hz from the cpufreq-info command.

**Comment by workhorsy** [comment]
@jonhp Can you try the beaglebone branch and tell me the output please?

```bash
python cpuinfo/cpuinfo.py
```

**Comment by jonhp** [comment]
debian@beaglebone:~/Downloads/test3/py-cpuinfo-master$ python cpuinfo/cpuinfo.py 
py-cpuinfo currently only works on X86 CPUs.debian@beaglebone:~/Downloads/test3/py-cpuinfo-master$

**Comment by workhorsy** [comment]
It looks like you are running the "master" branch. Try the "beaglebone" branch:
```bash
git pull
git checkout beaglebone
python cpuinfo/cpuinfo.py
```

**Comment by jonhp** [comment]
up until now i have been downloading the .zip files from github...
ok...i am in over my head in git...
git init py-cpuinfo
cd py-cpuinfo
git pull git://github.com/workhorsy/py-cpuinfo.git
git checkout beaglebone <><><>file not match error msg
git checkout -b beaglebone <><><>switched to new branch
cpuinfo.py <><><>i think it is still the one from master

**Comment by workhorsy** [comment]
Yeah. Learning Git was a pain in the butt.

You can download the zip from the branch page:

https://github.com/workhorsy/py-cpuinfo/archive/beaglebone.zip

**Comment by jonhp** [comment]
debian@beaglebone:~/Downloads/test5/py-cpuinfo-beaglebone$ python cpuinfo/cpuinfo.py 
Vendor ID: 
Brand: ARMv7 Processor rev 2 (v7l)
Hz Advertised: 1.0000 GHz
Hz Actual: 1.0000 GHz
Hz Advertised Raw: (1000000000, 0)
Hz Actual Raw: (1000000000, 0)
Arch: ARM_7
Bits: 32
Count: 1
Raw Arch String: armv7l
L2 Cache Size: 
L2 Cache Line Size: 0
L2 Cache Associativity: 0
Stepping: 0
Model: 0
Family: 0
Processor Type: 0
Extended Model: 0
Extended Family: 0
Flags: edsp, fastmult, half, neon, swp, thumb, thumbee, tls, vfp, vfpv3

**Comment by workhorsy** [comment]
That looks like everything we can get from ARM. Many of those fields are X86 only. Will that work for your BeagleBone?

**Comment by jonhp** [comment]
how about the hardware field?
debian@beaglebone:/var/lib/cloud9$ cat /proc/cpuinfo
processor	: 0
model name	: ARMv7 Processor rev 2 (v7l)
BogoMIPS	: 993.47
Features	: swp half thumb fastmult vfp edsp thumbee neon vfpv3 tls 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x3
CPU part	: 0xc08
CPU revision	: 2

Hardware	: Generic AM33XX (Flattened Device Tree)
Revision	: 0000
Serial		: 0000000000000000

**Comment by workhorsy** [comment]
Does this do everything?

https://github.com/workhorsy/py-cpuinfo/archive/beaglebone.zip

**Comment by jonhp** [comment]
hi,
I think it is a winner. I tried it out on rpiv2 as well as bbb. Data looks good on both platforms. 
Thanks.

**Comment by workhorsy** [comment]
Great! So I'm assuming that the PI has cpufreq-info too. Awesome.

**Comment by workhorsy** [comment]
Fixed in b22bac0cee7102ec1158dc7a8d783601962841f6
